### DISCUSSION TITLE
Managing code changes between STIG updates

### DISCUSSION THREAD

**Discussion by potatoqualitee** [mitre/saf#278] [Q&A] [answered]
I am currently working on converting the SQL Server xccdf to inspec.

Based off of what I've seen from [VMware's inspec profiles](https://github.com/vmware/dod-compliance-and-automation/tree/master/vsphere/7.0/vcsa/inspec/vmware-vcsa-7.0-stig-baseline), it appears I'll have to append some code at the end of many of the .rb files. For instance (just sample code, I don't know Ruby yet)

```ruby
  command = "(Get-DbaLogin -SqlInstance #{input('sqlInstance')} | Where-Object Id -eq 1).Name"
  describe ps_command(command) do
    its('stdout.strip') { should_not be 'sa' }
```

But the stig is often updated so I imagine I'd be regenerating my .rb files quarterly.

```
saf generate:xccdf2inspec_stub -o C:\temp\sqlstub --input="$home\Downloads\U_MS_SQL_Server_2016_Instance_STIG_V2R6_Manual-xccdf.xml"
```

How can I associate code that is not overwritten but the text is updated? I tried the metadata file but that doesn't seem to be what I'm after.

**Comment by camdenmoors** [comment, marked as answer]
Hi @potatoqualitee,

This is something we have started work on, however due to the prioritization of other projects it hasn't had much progress (see #94). 

While we do hope to have such feature in the future, it's currently a backlog item and not possible with the current version of the SAF CLI.

**Comment by potatoqualitee** [comment]
Ah, thank you! I will wait 👍🏼
